### ISSUE TITLE
[4.0] [a11y] Number of positions selector is not accessible

### ISSUE THREAD

**Issue by zwiastunsw** [joomla/joomla-cms#24470]
### Steps to reproduce the issue
1. Use screen reader, e.g. NVDA
2. Go to the search results list page (for ex. use search module to search for a popular word, e.g. joomla on a test site)
3. Go to "Display #" list.
4. Select any number position

### Expected result
1. The screen reader announces the selected number of items
2. After pressing the Enter key, the page is reloaded and the selected number of items is displayed.

### Actual result
1. The screen reader not announces the selected number
2. After a while, the page is automatically reloaded.

**Comment by hardik-codes** [comment]
Working on it

**Comment by gkrn** [comment]
i tried reproducing this issue, in my case the "Display # list " announces blank . Is this what you mean?

**Comment by brianteeman** [comment]
@zwiastunsw it is not just this field. It is every select field throughout joomla that uses the onchange event

As soon as you remove that then the list will be announced correctly.
BUT then there isn't a submit button so nothing will ever change

It's a big a11y problem and its everywhere in joomla

I don't know of a technique to resolve this but one must exist

**Comment by brianteeman** [comment]
More info and a better explanation of the problem

https://adamsilver.io/articles/select-boxes-shouldnt-submit-on-change/

**Comment by brianteeman** [comment]
As far as I can see there are only two possible solutions

1. Add a submit button
2. Use the javascript technique shown here http://www.themaninblue.com/experiment/AccessibleSelect/ which When you change the select using the keyboard it shouldn't submit its change until you select a value –	by pressing enter or tabbing off it.
The code for that is here http://www.themaninblue.com/experiment/AccessibleSelect/scripts/select_onchange.js
But the code is very old.

**Comment by brianteeman** [comment]
@wilsonge your thoughts on this please as it is throughout joomla and not just here

**Comment by brianteeman** [comment]
Sorry I was wrong suggesting that it was everywhere that we use onchange that is a problem.

If you go to the filter list in article manager. and use the keyboard to highlight a select list and try to cycle through the options using the arrow key it will fail. But if you press enter to select the list and open it then you can use the keyboard to navigate the list and select the option you want. I believe the same is true on the example @zwiastunsw gave

**Comment by wilsonge** [comment]
So it is technically a11y compliant but not as easy as it could/should be if I read that correctly?

**Comment by brianteeman** [comment]
That is my amended conclusion. not sure what @zwiastunsw thinks though

**Comment by zwiastunsw** [comment]
We have several places where the "Display select" object is displayed.  
We have two different behaviors.

**First behavior**: The screen reader announces a label ("Display # combo box 30 collapsed"). It announces that the list is collapsed. I get the information that I can expand it, select one of the options, and then confirm with the Enter key. This is the correct, accessible, expected behavior. (just like in the backend, as @brianteeman says). 
See, for example, view of Article Category List.
![article_category_list](https://user-images.githubusercontent.com/29527200/56287433-9a775080-611c-11e9-9722-84854f4ab9c5.png)
**Second behavior**: The screen reader does not advertise the label. The message is 'Edit read only'. I don't know that I have a combo box. I don't know that I can choose an option. As a sighted person I cannot expand the list with the Enter key. When I move the arrows around the list, the page is automatically reloaded. This is incorrect, inaccessible, unexpected behavior.  
See for example, view of Search results
![search](https://user-images.githubusercontent.com/29527200/56287766-841dc480-611d-11e9-8b2b-fc21943d4525.png)

**Comment by brianteeman** [comment]
found it and fixed it - pr coming

for some STUPID reason this list box was using chzn. That is not accessible as we know and we dont need it here

**Comment by brianteeman** [comment]
It was any select list in the search component that was infected with chosen

**Comment by zwiastunsw** [comment]
@brianteeman: The same problem we have in Tagged items view (joomla 3.9)

![tagged](https://user-images.githubusercontent.com/29527200/56293653-ca792080-6129-11e9-966b-4356aacd250b.png)

**Comment by brianteeman** [comment]
wait - you are confusing me - you raised this issue as Joomla 4

**Comment by brianteeman** [comment]
I fixed it for j3. And I cant get search to work for me at all right now

**Comment by zwiastunsw** [comment]
I'm sory

**Comment by brianteeman** [comment]
so are there any problems in joomla 4? We dont use chosen in j4

**Comment by zwiastunsw** [comment]
Looks like it's in the Joomla4 no problem. I checked it again.

**Comment by zwiastunsw** [comment]
Thank you, @brianteeman . I must have been very tired. This problem occurs only in joomla 3.x.

**Comment by brianteeman** [comment]
i fixed com_tags as well
